### DISCUSSION TITLE
Conditional if variable == something

### DISCUSSION THREAD

**Discussion by shanehughes1990** [fastlane/fastlane#16879] [Q&A] [answered]
Hello everyone, I am working on building out my fastlane file, I need a conditional if statement
here is what I got
.env.android has a variable called GITHUB_RELEASE_CHECK

I need to run different parts of a lane depending on what that variable is. IE GITHUB_RELEASE_CHECK=pre_release then run the puts "pre_release" section

```
desc "APK release pipeline"
lane :apk_release do
  Dir.chdir ".." do
    release_check = ENV['GITHUB_RELEASE_CHECK']
    puts release_check
    if release_check == release
      puts "release"
      if release_check == pre_release
      puts "pre_release"
      if release_check == skip_release
      puts "skip_release"
    end
  end
end
```

Apologies I am learning ruby because of fastlane so this is new to me

**Comment by tmxkevingriesbach** [comment, marked as answer]
You nested your ifs.  

Your code says this:
```
If release_check is release, then puts "release", then check if release_check is "pre_release",
  then puts "pre_release", then check if release_check is "skip_release",
    then puts "skip_release", endif
  endif
endif
```
You should check out how ruby if works here: https://www.tutorialspoint.com/ruby/ruby_if_else.htm

In short, you'll want this:
```ruby
puts release_check
if release_check == release
  puts "release"
elsif release_check == pre_release
  puts "pre_release"
elsif release_check == skip_release
  puts "skip_release"
end
```
Also, if release, pre_release and skip_release are strings and not variables, you'll need to add quotation marks around them.

**Comment by shanehughes1990** [comment]
You sir saved me so much headache. That worked like a charm, I have literally put 2 days in of ruby learning, so this is very new, thank you very much for getting back to me quickly. Now I know how to switch lanes conditionally sweet 👍 (And maybe some other new comers some headache)

for anyone else
.env.android (Our CI changes this variable during build time based on PR labels)
```
# Github release label
GITHUB_RELEASE_CHECK=release OR pre_release OR skip_release
```
Fastfile
```
platform :android do
  # Import .env.android environment
  before_all do
    Dotenv.load ".env.android"
  end

  desc "APK release pipeline"
  lane :apk_release do
    Dir.chdir ".." do
      release_check = ENV['GITHUB_RELEASE_CHECK']
      puts release_check
      if release_check == "release"
        puts "release"
      elsif release_check == "pre_release"
        puts "pre_release"
      elsif release_check == "skip_release"
        puts "skip_release"
      end
    end
  end
end
```
